tree/30_generic_methods/permissions_group_acl_absent.cf

```diff
@@ -0,0 +1,218 @@
+      "path_exists_${canonified_file[${indices}]}"    expression => fileexists("${file_list}");
+
+    pass2.files_found.!is_recursive::
+      # Do not forget the ":"
+      "need_change_${canonified_file[${indices}]}" expression => returnszero("${paths.getfacl} -c ${file[${indices}]} 2>/dev/null | ${paths.grep} -q \"${group_ace}:\" #Initial Check", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}";
+
+    pass2.files_found.is_recursive::
+      # Do not forget the ":"
+      "need_change_${canonified_file[${indices}]}" expression => returnszero("${paths.getfacl} -Rc ${file[${indices}]} 2>/dev/null | ${paths.grep} -q \"${group_ace}:\" #Initial Check", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}";
+
+    pass2.files_found.!(global_dry_run|dry_run).!is_recursive::
+      "remove_acl_${canonified_file[${indices}]}"  expression => returnszero("${paths.setfacl} -x g:${group} ${file[${indices}]}", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}.need_change_${canonified_file[${indices}]}";
+
+      "change_verified_${canonified_file[${indices}]}"    not => returnszero("${paths.getfacl} -c ${file[${indices}]} 2>/dev/null | ${paths.grep} -q \"${group_ace}:\" #Post repaired Check", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}.need_change_${canonified_file[${indices}]}";
+
+    pass2.files_found.!(global_dry_run|dry_run).is_recursive::
+      "remove_acl_${canonified_file[${indices}]}"  expression => returnszero("${paths.setfacl} -Rx g:${group} ${file[${indices}]}", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}.need_change_${canonified_file[${indices}]}";
+
+      "change_verified_${canonified_file[${indices}]}"    not => returnszero("${paths.getfacl} -Rc ${file[${indices}]} 2>/dev/null | ${paths.grep} -q \"${group_ace}:\" #Post repaired Check", "useshell"),
+        ifvarclass => "path_exists_${canonified_file[${indices}]}.need_change_${canonified_file[${indices}]}";
+
+    pass2.files_found::
+      "inner_class_${indices}_kept"     expression => "!need_change_${canonified_file[${indices}]}";
+      "inner_class_${indices}_repaired" expression => "need_change_${canonified_file[${indices}]}.change_verified_${canonified_file[${indices}]}";
+      "inner_class_${indices}_error"    expression => "!inner_class_${indices}_kept.!inner_class_${indices}_repaired";
+
+      "merging_failure"  expression => "inner_class_${indices}_error";
+      "merging_repaired" expression => "inner_class_${indices}_repaired";
+
+
+  methods:
+    pass3.!files_found::
+      "No_file_found" usebundle =>  _classes_failure("${class_prefix}");
+      "No_file_found" usebundle =>  _classes_failure("${old_class_prefix}");
+      "report"        usebundle => _log_v3("No files could be found matching ${path}, remove POSIX acl for group ${group}", "${path}", "${old_class_prefix}", "${class_prefix}", @{args});
```

Comment:
the report is a bit misleasding, what about
No files could be found matching ${path} when trying to remove POSIX acl for group ${group}

Reply:
Since we are adding a suffix "could not be repaired" the original sentence may be better